rcgen/src/certificate.rs

```diff
@@ -461,6 +467,10 @@ impl CertificateParams {
 	}
 
 	fn write_subject_alt_names(&self, writer: DERWriter) {
+		if self.subject_alt_names.is_empty() {
```

Comment:
this is actually required by the spec, it seems:

```
If the subjectAltName extension is present, the sequence MUST contain
   at least one entry.
```

Reply:
Nice! If I'm understanding correctly, that means that the current behavior—which allows a sequence with zero entries—is incorrect, and the code addition you highlighted is correct.

Reply:
I refactored this into [a4f8b94](https://github.com/rustls/rcgen/pull/296/commits/a4f8b94675fb0df45ed2ab3638113bbe52e093a4)